primary/src/metrics.rs

```diff
@@ -16,35 +16,23 @@ pub(crate) struct Metrics {
     pub(crate) node_metrics: Option<PrimaryMetrics>,
 }
 
-static mut METRICS: Metrics = Metrics {
-    endpoint_metrics: None,
-    primary_endpoint_metrics: None,
-    node_metrics: None,
-};
-static INIT: Once = Once::new();
```

Comment:
No need to keep the metrics behind a `Once` - since we can pass a Registry we can always pass a separate Registry per test so we don't bump into conflict issues on the default registry